app/UI/message-panel.tsx

```diff
@@ -204,81 +205,103 @@ export function ParseMessageContent(
     const parsedContentItems = parseContent(message.content);
 
     const vnode = [];
-    let start = 0;
     for (const item of parsedContentItems) {
-        vnode.push(message.content.slice(start, item.start));
-        const itemStr = message.content.slice(item.start, item.end + 1);
-        if (item.type == "url") {
-            if (urlIsImage(itemStr)) {
+        if (item.type === "normal") {
+            vnode.push(item.text);
+        } else if (item.type == "url") {
+            if (urlIsImage(item.text)) {
                 vnode.push(
                     <img
                         class={`w-96 p-1 rounded-lg border-2 border-[${DividerBackgroundColor}]`}
-                        src={itemStr}
+                        src={item.text}
                     />,
                 );
-            } else if (urlIsVideo(itemStr)) {
+            } else if (urlIsVideo(item.text)) {
                 vnode.push(
                     <video
                         class={`w-96 p-1 rounded-lg border-2 border-[${DividerBackgroundColor}]`}
                         controls
-                        src={itemStr}
+                        src={item.text}
                     >
                     </video>,
                 );
             } else {
                 vnode.push(
-                    <a target="_blank" class={`hover:underline text-[${LinkColor}]`} href={itemStr}>
-                        {itemStr}
+                    <a target="_blank" class={`hover:underline text-[${LinkColor}]`} href={item.text}>
+                        {item.text}
                     </a>,
                 );
             }
         } else if (item.type == "npub") {
-            const profile = getters.profileGetter.getProfileByPublicKey(item.pubkey);
+            const bech32 = item.text.startsWith("nostr:") ? item.text.slice(6) : item.text;
+            const pubkey = PublicKey.FromBech32(bech32);
+            if (pubkey instanceof Error) continue;
```

Comment:
Move the construction of PublicKey to the parser.

As a general rule, when you find yourself ignoring errors, it's a sign of wrong design and/or implementation

Reply:
I think the construction of PublicKey and Render is related, and there is no need to put this part of the logic in the parseContent function.

Putting the part of building the public key into parseContent will make it more complicated to test parseContent.

Reply:
If constructing the public key fails, how should it be handled? Previously, the approach was to omit this fragment.

However, if we view parseContent as dividing the incoming content based on regex, the absence of a part due to an error complicates its interpretation.